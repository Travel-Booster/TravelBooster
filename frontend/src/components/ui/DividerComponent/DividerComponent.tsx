import React, { memo } from 'react'
import { View } from 'react-native'
import styles from './DividerStyles'

/**
 * Divider Component
 * @component
 * @description Divider Component
 * @param {any} size - divider size, default: 'medium' (optional props)
 * @returns {object} - return component with children
 */
const DividerComponent = memo<DividerComponentProps>(
	({ size = 'medium' }) => <View style={styles[size]} />,
)

/**
 * Props types
 */
type DividerComponentProps = {
	size?: Sizes
}

/**
 *  Size variant
 */
type Sizes = 'tiny' | 'small' | 'medium' | 'big'

export default DividerComponent
