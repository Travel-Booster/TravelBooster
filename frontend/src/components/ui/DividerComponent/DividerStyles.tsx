import { StyleSheet } from 'react-native'
import paddings from 'styles/paddings'

export default StyleSheet.create({
  tiny: {
    height: paddings.TINY
  },
  small: {
    height: paddings.SMALL
  },
  medium: {
    height: paddings.MEDIUM
  },
  big: {
    height: paddings.BIG
  }
})
