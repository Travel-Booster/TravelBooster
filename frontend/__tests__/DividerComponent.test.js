import DividerComponent from 'components/ui/DividerComponent'
import React from 'react'
import renderer from 'react-test-renderer'

const props = {
	size: 'small',
}

const component = <DividerComponent {...props} />
const wrapper = renderer.create(component).toJSON()

it('<TextComponent /> renders', () => {
	expect(wrapper).toBeTruthy()
})

it('<TextComponent /> test snapshot', () => {
	expect(wrapper).toMatchSnapshot()
})
