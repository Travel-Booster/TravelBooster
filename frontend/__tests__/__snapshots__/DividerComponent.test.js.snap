// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent /> test snapshot 1`] = `
<View
  style={
    Object {
      "height": 10,
    }
  }
/>
`;
